### PR TITLE
Improve screencast performance by switching to OSX AVFoundation screen capture

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
I've tested the changed server and the original one built from master and a significant improvement in FPS performance can be observed. Full screen video plays at 25-30 FPS, windowed - 12-25 FPS. 

~~Although, the normal interaction with the mouse cursor has a tiny lag. I don't think this could be easily workarounded, since the captured video frames are already delayed from what is happening in real time.~~

This was fixed after switching to `CGDisplayStream`

### CODE REVIEW

File: OSXvnc-server/AVSampleBufferHolder.m [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//  Created by Mykola Mokhnach on 19.01.19.
+//  Copyright (c) 2019 Sauce Labs Inc. All rights reserved.
+//
+
+#import "AVSampleBufferHolder.h"
+
+
+@implementation AVSampleBufferHolder
+
+@synthesize timestamp = _timestamp;
```
Comment (dmissmann) [inline OSXvnc-server/AVSampleBufferHolder.m]: That's the default synthesis, can be removed.

File: OSXvnc-server/AVSampleBufferHolder.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+
+#import <Foundation/Foundation.h>
+#import <AVFoundation/AVFoundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface AVSampleBufferHolder : NSObject
+
+@property (assign, nonatomic, nullable) CMSampleBufferRef sampleBuffer;
```
Comment (dmissmann) [inline OSXvnc-server/AVSampleBufferHolder.h]: `assign` is not correct here. You call retain in the setter.

**Review by zwanesl — APPROVED** [review]
👊🏿which vnc client did you test with?

**Comment by mykola-mokhnach** [comment]
@zwanesl I tested with noVNC web client and TurboVNC for Mac client

File: OSXvnc-server/AVSampleBufferHolder.h (hunk 2 above)
Comment (mykola-mokhnach) [reply OSXvnc-server/AVSampleBufferHolder.h]: here we have our own overridden variable setter. There is no need to tell the compiler to do any additional actions.

File: OSXvnc-server/AVSampleBufferHolder.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
+//
+//  AVSampleBufferHolder.h
+//  OSXvnc
+//
+//  Created by Mykola Mokhnach on 19.01.19.
+//  Copyright (c) 2019 Sauce Labs Inc. All rights reserved.
```
Comment (stweil) [inline OSXvnc-server/AVSampleBufferHolder.h:6]: OSXvnc uses GPL 3. This copyright statement violates that license. Could you please fix it?

**Pull request merged**